app/frontend/controllers/event_controller.js

```diff
@@ -0,0 +1,136 @@
+/* This file is part of Ezra Bible App.
+
+   Copyright (C) 2019 - 2021 Ezra Bible App Development Team <contact@ezrabibleapp.net>
+
+   Ezra Bible App is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   Ezra Bible App is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Ezra Bible App. See the file LICENSE.
+   If not, see <http://www.gnu.org/licenses/>. */
+
+
+/**
+ * This controller implements pub/sub pattern and serves as a broker (event bus) between
+ * event emitter (publisher) and event consumer subscriber.
+ * https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern
+ * 
+ * @module eventController
+ * @category Controller
+ */
+
+var messages = {};
```

Comment:
Why `messages` and not `subscribers`?

Reply:
My initial thought was that publisher delivers _messages_ to subscribers according to the common pattern.
But this variable hold subscribers and not messages. So I'll change that.